lib/logreader.c

```diff
@@ -231,7 +231,12 @@ log_reader_apply_proto_and_poll_events(LogReader *self, LogProtoServer *proto, P
   self->proto = proto;
 
   if (self->proto)
-    log_proto_server_set_wakeup_cb(self->proto, (LogProtoServerWakeupFunc) log_reader_wakeup, self);
+    {
+      log_proto_server_set_wakeup_cb(self->proto, (LogProtoServerWakeupFunc) log_reader_wakeup, self);
+      self->handshake_in_progress = log_proto_server_needs_handshake(self->proto);
```

Comment:
why do we need to reset handshake_in_progress? the idea was to always start in "handshake" mode and the handshake method returns when handshake finishes. we should not need to check that again, at least in my opinion.

Reply:
I've just checked, handshake_finished is only set to TRUE if either the LogProto claims it has finished the handshake, or when we switch the a LogProto that does not have a handshake method.